src/Adapter/MSTestAdapter.PlatformServices/Execution/TestMethodInfo.cs

```diff
@@ -671,13 +671,22 @@ private static TestFailedException HandleMethodException(Exception ex, Exception
     /// <param name="executionContext">The execution context to run on.</param>
     /// <param name="timeoutTokenSource">The timeout token source.</param>
     [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
-    private void RunTestCleanupMethod(TestResult result, ExecutionContext? executionContext, CancellationTokenSource? timeoutTokenSource)
+    private
+#if NET6_0_OR_GREATER
+        async
+#endif
+        Task
```

Comment:
why this? (too stupid to figure it out myself :D)

Reply:
We could keep it `void` on non-net6.0, but then callers need to be conditional as well.